### PR TITLE
Removes quotes around vars for auths - fixes #2784

### LINKED ISSUE #2784: PlannerTask: Quotes not removed around ApplicationId, CertificateThumbprint, TenantId etc
#### Details of the scenario you tried and the problem that is occurring
Previously, only *Credential* auth was supported for Planner, therefore, in `MSFT_PlannerTask.psm1`, function `Export-TargetResource`, line 787: 

```powershell
$content = Convert-DSCStringParamToVariable -DSCBlock $content `
    -ParameterName 'Credential'
```
ensures that quotes are removed around `$ConfigurationData.NonNodeData.Credential`: 
```powershell
PlannerTask 7ad38863-ce48-4b8e-b169-c292222471e6
{
    Bucket = "nYij0yLaMJ-KKPkQnX1wGpcAMF4C";
    Categories = @("Blue");
    Credential = $ConfigurationData.NonNodeData.Credential;
    DueDateTime = "15.05.2019 10:00:00";
    Ensure = "Present";
    PercentComplete = 100;
    PlanId = "j0P6vPeqpUiI2UUlwClBjZcAHXJ4";
    Priority = 5;
    TaskId = "SC81Z9zF9EGMlI76hYauj5cAD8o4";
    Title = "Some Title";
}
```
However, if for instance if Certificate is used for auth, quotes are *not* removed for properties related to that (i.e. *CertificateThumbPrint*, *ApplicationId*, *TenantId*): 
```powershell
PlannerTask 7ad38863-ce48-4b8e-b169-c292222471e6
{
    # ApplicationId will fail because of the quotes below
    ApplicationId = "$ConfigurationData.NonNodeData.ApplicationId";
    Bucket = "nYij0yLaMJ-KKPkQnX1wGpcAMF4C";
    Categories = @("Blue");
    # CertificateThumbprint will fail because of the quotes below
    CertificateThumbprint = "$ConfigurationData.NonNodeData.CertificateThumbprint";
    DueDateTime = "15.05.2019 10:00:00";
    Ensure = "Present";
    PercentComplete = 100;
    PlanId = "j0P6vPeqpUiI2UUlwClBjXcAHXJ4";
    Priority = 5;
    TaskId = "SC91Z8zF9EGMlI76hYauj5cAD8o4";
    # TenantId will fail because of the quotes below
    TenantId = "$ConfigurationData.NonNodeData.TenantId";
    Title = "Some task";
}
```
#### Verbose logs showing the problem
n/a
#### Suggested solution to the issue

Change the following in `MSFT_PlannetTask.psm1`, function `Export-TargetResource`, starting at line 787:
```powershell
$content = Convert-DSCStringParamToVariable -DSCBlock $content `
    -ParameterName 'Credential'
```
with the following lines: 
```powershell
# Ensure quotes are removed around auth variables
if ($Credential) 
{
    $content = Convert-DSCStringParamToVariable -DSCBlock $content `
        -ParameterName 'Credential'
}
if ($ApplicationId) 
{
    $content = Convert-DSCStringParamToVariable -DSCBlock $content `
        -ParameterName 'ApplicationId'
}
if ($ApplicationSecret) 
{
    $content = Convert-DSCStringParamToVariable -DSCBlock $content `
        -ParameterName 'ApplicationSecret'
}
if ($TenantId) 
{
    $content = Convert-DSCStringParamToVariable -DSCBlock $content `
        -ParameterName 'TenantId'
}
if ($CertificateThumbprint) 
{
    $content = Convert-DSCStringParamToVariable -DSCBlock $content `
        -ParameterName 'CertificateThumbprint'
}
```

#### The DSC configuration that is used to reproduce the issue (as detailed as possible)
just exporting

#### The operating system the target node is running
OsName               : Microsoft Windows 10 Enterprise
OsOperatingSystemSKU : EnterpriseEdition
OsArchitecture       : 64-bit
WindowsVersion       : 2009
WindowsBuildLabEx    : 19041.1.amd64fre.vb_release.191206-1406
OsLanguage           : en-US
OsMuiLanguages       : {en-US, nb-NO}

#### Version of the DSC module that was used ('dev' if using current dev branch)
Dev

### PR DESCRIPTION
**Pull request by bjoernf73**
#### Pull Request (PR) description
Removes quotes around vars for auths, see #2784 

#### This Pull Request (PR) fixes the following issues
- Fixes #2784

### CODE REVIEW

File: Modules/Microsoft365DSC/DSCResources/MSFT_PlannerTask/MSFT_PlannerTask.psm1 [hunk 1]
```diff
@@ -784,8 +784,34 @@ function Export-TargetResource
                         $currentDSCBlock += '        PlannerTask ' + (New-Guid).ToString() + "`r`n"
                         $currentDSCBlock += "        {`r`n"
                         $content = Get-DSCBlock -Params $result -ModulePath $PSScriptRoot
-                        $content = Convert-DSCStringParamToVariable -DSCBlock $content `
-                            -ParameterName 'Credential'
+
```
Comment (andikrueger) [inline Modules/Microsoft365DSC/DSCResources/MSFT_PlannerTask/MSFT_PlannerTask.psm1:787]: Please have a look at these two functions. They are used in other resources and do combine the steps below:

$Results = Update-M365DSCExportAuthenticationResults -ConnectionMode $ConnectionMode `
                    -Results $Results
                $currentDSCBlock = Get-M365DSCExportContentForResource -ResourceName $ResourceName `
                    -ConnectionMode $ConnectionMode `
                    -ModulePath $PSScriptRoot `
                    -Results $Results `
                    -Credential $Credential

**Review by andikrueger — CHANGES_REQUESTED** [review]
Please have a look at the comments and add a note to the changelog. Thanks!

File: Modules/Microsoft365DSC/DSCResources/MSFT_PlannerTask/MSFT_PlannerTask.psm1 (hunk 1 above)
Comment (bjoernf73) [reply Modules/Microsoft365DSC/DSCResources/MSFT_PlannerTask/MSFT_PlannerTask.psm1]: @andikrueger Fixed in latest commit.

**Comment by bjoernf73** [comment]
@andikrueger Should be resolved now.

**Review by andikrueger — COMMENTED** [review]
LGTM

**Pull request merged**